web/src/composebox_typeahead.ts

```diff
@@ -644,27 +820,38 @@ export function get_sorted_filtered_items(query, input_element) {
 
     const big_results = get_candidates(query, input_element);
 
-    if (!big_results) {
+    if (big_results === false) {
         return [];
     }
 
-    const opts = get_stream_topic_data(input_element);
-
-    if (completing === "mention" || completing === "silent_mention") {
-        return filter_and_sort_mentions(big_results.is_silent, token, opts);
+    // These are sorted separately
+    if (big_results.type === "mention") {
+        return big_results.list;
     }
 
-    return filter_and_sort_candidates(completing, big_results, token);
+    assert(big_results.type === completing);
+    return filter_and_sort_candidates(big_results, token);
 }
 
-export function filter_and_sort_candidates(completing, candidates, token) {
-    const matcher = compose_content_matcher(completing, token);
-
-    const small_results = candidates.filter((item) => matcher(item));
-
-    const sorted_results = sort_results(completing, small_results, token);
-
-    return sorted_results;
+export function filter_and_sort_candidates<
+    T extends
+        | TopicJumpList
+        | TimeJumpList
+        | LanguageList
+        | TopicList
+        | EmojiList
+        | SlashCommandList
+        | StreamList,
+>(candidates: T, token: string): TypeaheadSuggestion[] {
+    const filtered_candidates = candidates;
```

Comment:
Is this extra assignment an attempt to avoid mutating `candidates`? It doesn’t work; `candidates` is still mutated through this reference.

Reply:
Oh, no this is just remains from when I was futzing around with various things trying to get TypeScript working. It doesn't need to be here.